Fix ID on labels

smithdc1 opened the pull request:
This PR fixes IDs for checkbox labels. 

Some discussion on this at #816. I think what I spotted was a different issue to the end of that discussion. 816 was referring to `id_id`. The code I spotted that was wrong was outputting `id__#` (double underscore), my fix now outputs `id_name_#`

**Old:**
![Capture9](https://user-images.githubusercontent.com/39445562/65283792-dd0be180-db2f-11e9-957d-6409ae362981.PNG)

**New:**
![Capture10](https://user-images.githubusercontent.com/39445562/65283793-dd0be180-db2f-11e9-93b7-e804ce91775a.PNG)

@carltongibson do you think it's worth adding a test to check for the format of ids? (not sure how hard this will be, but am willing to give it a go!

EDIT: 

Oops - forgot about the tests so it looks like there is already something in place. Will try and look at tomorrow.

smithdc1 commented:
Oops - I forgot about the tests, looks like they are failing. One for another day....

carltongibson commented:
Yep. Failing test is `TestBootstrapLayoutObjects.test_multiple_checkboxes_unique_ids[bootstrap4] __`

carltongibson commented:
At a guick glance, it looks like we **should** be using `id_for_label` to get the right `id`. 

See https://github.com/django/django/blob/4a954cfd11a5d034491f87fcbc920eb97a302bb3/docs/ref/forms/widgets.txt#L277

What happens if we just drop the additional `id_` prefix?

Should we adjust the related changes to use that? 

(Rather that switching to `html_name` throughout, as the suggestions had... 😬)

Thanks for the effort @smithdc1!

smithdc1 commented:
I think this needs to be improved to a holistic 'id fix' PR

- change labels to `id_for_label` 
- check no id_id tags
- check no blank tags (i.e. double underscore)
- pass / update tests as required


- 

-

smithdc1 commented:
Waiting for #897 to land and then I'll update this, as a few of these lines are changing. I've done a bit more work on it so hopefully it will just need an odd tweak a bit of checking in due course.

smithdc1 commented:
Hi @carltongibson or @bryan-brancotte ,

This is ready for review. 

- Removed `ID=` for `<label>` on multiple-checkbox. This style is now consistent with all the other labels not having an `id` tag.
- Updated code to use `id_for_label` for `checkboxselectmultiple_inline.html` and `radioselect_inline.html`
- Changed code to create `for=` in radios to match checkbox. The output now matches the `id=` of the `<input>`. It is now generating the format `id_checkboxes_1`, previously it was `id_checkboxes_0_1`
